projects/hslayers/src/components/add-data/common/url/layer-table/layer-table.component.html

```diff
@@ -0,0 +1,118 @@
+<div>
+    <ng-container *ngIf="type === 'wms'; else simpleTable">
+        <table class="table table-sm table-striped table-bordered hs-url-table">
+            <thead class="bg-secondary text-light">
+                <th style="width:1.5em">&nbsp;</th>
+                <th>{{'COMMON.title' | translateHs: {app} }}</th>
+                <!-- <th style="width:30%">{{'COMMON.name' | translateHs: {app} }}</th> -->
+                <th>{{'COMMON.abstract' | translateHs: {app} }}</th>
+            </thead>
+            <tbody *ngFor="let layer of data.layers | slice:0:limitShown; trackBy:( 'Name' | trackByProperty )">
+                <tr>
+                    <td [attr.rowspan]="layer.Style?.length ? 3 : 2"
+                        [ngClass]="{'align-middle': !hasNestedLayers(layer)}">
+                        <input type="checkbox" class="checkbox-lg" [(ngModel)]="layer.checked" name="layer"
+                            style="cursor: pointer;" (change)="searchForChecked(layer)" />
+                    </td>
+                    <td hsWmsLayerHighlight style="max-width: 25.375em" class="text-truncate"
+                        [title]="'Title: ' + layer.Title  + '\nName: ' + layer.Name"
+                        (click)="hsUrlWmsService.expandTableRow($event)">{{layer.Title}}</td>
+                    <!-- <td style="max-width: 13.75em" class="text-truncate" [title]="layer.Name">{{layer.Name}}</td> -->
+                    <td hsWmsLayerHighlight style="max-width: 10.875em; width:30%" class="text-truncate"
+                        [title]="layer.Abstract" (click)="hsUrlWmsService.expandTableRow($event)">
+                        {{layer.Abstract}}</td>
+                </tr>
+                <tr *ngIf="layer.Style?.length > 1 && layer.checked">
+                    <td colspan="3">
+                        <div class="d-flex align-items-baseline px-2 w-100">
+                            <div class="d-flex align-items-baseline" style="flex-grow: 1;">
+                                <label class="control-label">{{'COMMON.style' | translateHs: {app} }}:</label>
+                                <select class="form-control form-select-sm form-select"
+                                    [(ngModel)]="layer.styleSelected" [ngModelOptions]="{standalone: true}">
+                                    <option *ngFor="let style of layer.Style" [ngValue]="style">{{style.Name}}
+                                    </option>
+                                </select>
+                            </div>
+                        </div>
+                    </td>
+                </tr>
+                <tr [hidden]="!hasNestedLayers(layer)">
```

Comment:
Appreciate the rewrite to table component. Maybe if you are at it, could you rewrite hasNestedLayers to be observable with async pipe or just a simple variable instead of a function? For performance reasons.

Reply:
Surely

Reply:
Was kinda straight forward. Function is declared in utils and used also in dimensions service but I think its ok in there so I left those intact